src/Parser/BrewmasterMonk/Modules/Spells/BlackoutCombo.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { SPELLS_WHICH_REMOVE_BOC } from '../../Constants';
+
+const debug = false;
+const BOC_DURATION = 15000;
+
+class BlackoutCombo extends Module {
+  blackoutComboConsumed = 0;
+  blackoutComboWasted = 0;
+  blackoutComboBuffs = 0;
+  lastBlackoutComboCast = 0;
+
+  on_initialized() {
+    this.active = this.owner.selectedCombatant.hasTalent(SPELLS.BLACKOUT_COMBO_TALENT.id);
+  }
+  
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (SPELLS.BLACKOUT_COMBO_BUFF.id === spellId) {
+      debug && console.log('Blackout combo applied');
+      this.blackoutComboBuffs++;
+      this.lastBlackoutComboCast = event.timestamp;
+    }
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+    if (SPELLS.BLACKOUT_COMBO_BUFF.id === spellId) {
+      debug && console.log('Blackout combo refreshed');
+      this.blackoutComboBuffs++;
+      this.blackoutComboWasted++;
+      this.lastBlackoutComboCast = event.timestamp;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (SPELLS_WHICH_REMOVE_BOC.indexOf(spellId) === -1) {
+      return;
+    }
+    // BOC should be up
+    if (this.lastBlackoutComboCast > 0 && this.lastBlackoutComboCast + BOC_DURATION > event.timestamp) {
+      this.blackoutComboConsumed++;
```

Comment:
This is all i had time last night and i wanted to get the PR created, that is next to add and will be in a PR today.